analysis/warriorfury/src/modules/shadowlands/covenants/SpearOfBastion.tsx

```diff
@@ -0,0 +1,37 @@
+import SPELLS from 'common/SPELLS';
+import COVENANTS from 'game/shadowlands/COVENANTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+class SpearOfBastion extends Analyzer {
+  _spearOfBastionDamage: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasCovenant(COVENANTS.KYRIAN.id);
+
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.SPEAR_OF_BASTION_ENERGIZE),
+      this.onSpearDamage,
+    );
+  }
+
+  onSpearDamage(event: DamageEvent) {
+    this._spearOfBastionDamage += event.amount;
```

Comment:
you should also look at event.absorbed so damage to shields is counted (as that is still effective damage)